src/asm/x86/predict.rs

```diff
@@ -48,208 +58,207 @@ decl_cfl_pred_fn!(rav1e_ipred_cfl_128_avx2);
 decl_cfl_pred_fn!(rav1e_ipred_cfl_left_avx2);
 decl_cfl_pred_fn!(rav1e_ipred_cfl_top_avx2);
 
-pub trait Intra<T>: native::Intra<T>
+pub trait Intra<T: Pixel>: native::Intra<T>
 where
   T: Pixel,
 {
-  fn pred_dc(output: &mut PlaneRegionMut<'_, T>, above: &[T], left: &[T]) {
-    {
-      if size_of::<T>() == 1 && is_x86_feature_detected!("avx2") {
-        return unsafe {
-          rav1e_ipred_dc_avx2(
-            output.data_ptr_mut() as *mut _,
-            output.plane_cfg.stride as libc::ptrdiff_t,
-            above.as_ptr().offset(-1) as *const _,
-            Self::W as libc::c_int,
-            Self::H as libc::c_int,
-            0,
-          )
-        };
-      }
+  fn pred_dc(
```

Comment:
I'd rather have a separate impl and pick the correct one where this is used.

Reply:
Could you clarify what you're looking for? I'm a bit confused.

Reply:
At what level should we be pickling the correct one? If we select it in `predict_intra_inner`, it makes the code messier and removes the isolation of the ASM functions, because of how we currently choose "asm" or "non-asm" at compile time by importing the correct crate. I don't believe we would be able to import the correct feature level at runtime, so we would need a different approach.

Reply:
Per IRC discussion, barrbrain will work on this in a separate PR.